chain/chain/src/resharding.rs

```diff
@@ -181,15 +202,49 @@ impl Chain {
         sync_hash: &CryptoHash,
         state_roots: Result<HashMap<ShardUId, StateRoot>, Error>,
     ) -> Result<(), Error> {
-        let prev_hash = *self.get_block_header(sync_hash)?.prev_hash();
+        let state_roots = state_roots?;
```

Comment:
Why is an input argument a `Result`? Seems to be simpler to check `is_ok()` at the callsite than immediately inside the function.

Reply:
Yeah, that makes sense. Mainly was there due to previous code I guess.
If we are asking why do we store a result a layer above, I guess it's because the error is generated in the `build_state_for_split_shards` function in sync job actor but we only want to expose it via the client actor in `build_state_for_split_shards_postprocessing` function.